Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1994,9 +2004,9 @@ extension BrowserViewController: TabManagerDelegate {
                 topToolbar.hideProgressBar()
             }
 
-            if tab.type != previous?.type {
-                let theme = Theme.of(tab)
-                applyTheme(theme)
+            let newTheme = Theme.of(tab)
+            if previous == nil || newTheme != Theme.of(previous) {
```

Comment:
Switched from using `tab.type` to using theme change detection. Also, will "force" update the theme if the previous tab didn't exist.